tiledb-sys/src/datatype.rs

```diff
@@ -207,6 +209,96 @@ impl Datatype {
             _ => None,
         }
     }
+
+    pub fn is_valid<T: 'static>(&self, _: &T) -> bool {
+        if TypeId::of::<T>() == TypeId::of::<f32>() {
```

Comment:
I think a tuple-type `match (self, TypeId::of::<T>())` could probably neaten this a bit.

Reply:
Not sure what you mean here. Self would be the enum value in this case which is just a variant of an enum rather than the data being converted.